### ISSUE TITLE
Vendure project created using NPX doesn't work well with Yarn's Workspace

### ISSUE THREAD

**Issue by macabeus** [vendure-ecommerce/vendure#480]
**Describe the bug**
I'm creating a project with Vendure and my own storefront. So I would like to use a single repository, sharing the dependencies using Yarn's Workspace.
But it isn't working when I added a custom authentication method, probably because it changes the GraphQL schemas. I receive the following error.

```
> yarn run start
yarn run v1.22.5
$ concurrently yarn:run:*
$ ts-node ./src/index-worker.ts
$ ts-node ./src/index.ts
[run:worker] info 9/28/20, 6:32 PM - [Vendure Worker] Bootstrapping Vendure Worker (pid: 73650)... 
[run:server] info 9/28/20, 6:32 PM - [Vendure Server] Bootstrapping Vendure Server (pid: 73649)... 
[run:worker] info 9/28/20, 6:32 PM - [NestMicroservice] Nest microservice successfully started 
[run:worker] info 9/28/20, 6:32 PM - [Vendure Worker] Vendure Worker started with TCP transport at localhost:3020 
[run:server] error 9/28/20, 6:32 PM - [ExceptionHandler] Invalid schema passed
[run:server] Error: Invalid schema passed
[run:server]     at /Users/macabeus/project/node_modules/graphql-tools/src/stitching/mergeSchemas.ts:197:13
[run:server]     at Array.forEach (<anonymous>)
[run:server]     at mergeSchemasImplementation (/Users/macabeus/project/node_modules/graphql-tools/src/stitching/mergeSchemas.ts:119:11)
[run:server]     at Object.mergeSchemas (/Users/macabeus/project/node_modules/graphql-tools/src/stitching/mergeSchemas.ts:77:10)
[run:server]     at Object.generateListOptions (/Users/macabeus/project/node_modules/@vendure/core/src/api/config/generate-list-options.ts:78:12)
[run:server]     at createTypeDefs (/Users/macabeus/project/node_modules/@vendure/core/src/api/config/configure-graphql-module.ts:175:18)
[run:server]     at createGraphQLOptions (/Users/macabeus/project/node_modules/@vendure/core/src/api/config/configure-graphql-module.ts:119:19) 
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

As workaround I needed to use the config [`nohoist`](https://classic.yarnpkg.com/blog/2018/02/15/nohoist/):

```json
"workspaces": {
  "packages": [
    "my_storefront",
    "my_vendure"
  ],
  "nohoist": ["**/@vendure/**"]
}
```

**To Reproduce**
1. Create a new project using Vendure and Yarn's Workspace
2. Everything will work fine at this moment. But now, add a custom authentication method, and try to run Vendure
3. See the error

**Expected behavior**
Vendure should work 

**Environment (please complete the following information):**
 - @vendure/core version: 0.15.1
 - Nodejs version: v12.15.0
 - Database: Postgres

**Additional context**
I would like to help to fix this bug, but I need some context to understand better what I should to do

**Comment by michaelbromley** [comment]
Hi,

can you show me the code of your custom auth strategy?

**Comment by macabeus** [comment]
@michaelbromley I created a minimal example to reproduce this bug: https://github.com/macabeus/vendure-issue-480

**Comment by michaelbromley** [comment]
Ok thanks for taking the time to make a repo for that. I strongly suspect that the reason is that you are using graphql v15 in your storefront, whereas Apollo Server (which Vendure uses) requires v14. So the hoisting is probably causing Vendure to use v15 when parsing the schema on bootstrap. You can test this hypothesis by using v14 in your storefront and turning hoisting back on.

**Comment by macabeus** [comment]
@michaelbromley I just tested it, doing a downgrade to `graphql` version `14.7.0` on storefront, but it doesn't solve the bug 😕

**Comment by michaelbromley** [comment]
hmm... ok. In the past I've run into problems with graphql-js even with the same major version, but multiple packages in the dependency tree, because internally in several functions it does an `instanceof` check on class instances, and this fails if the class if from a _different_ physical file (even if it's the exact same version of the lib). 

More info on this issue here: https://github.com/graphql/graphql-js/issues/491

You might just have to turn off hoisting for graphql, and allow it for the rest.

**Comment by macabeus** [comment]
I tried to use the following arguments on `nohoist`, but it doesn't work.
```
"nohoist": ["**/graphql.**"]
```
```
"nohoist": ["**/graphql/**"]
```

But it's look like that Yarn Workspace don't look on `nohoist` when is handling with dependencies of dependencies. I checked the `node_modules` folder on `vendure-app` and there is no `graphql` folder.

Anyway, it's a bug on `graphql`, not on vendure itself, so I think that this issue could be closed.
